src/Microsoft.Diagnostics.ExtensionCommands/FindPointersInCommand.cs

```diff
@@ -0,0 +1,517 @@
+﻿using Microsoft.Diagnostics.Runtime;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using static Microsoft.Diagnostics.ExtensionCommands.MAddressCommand;
+using Microsoft.Diagnostics.DebugServices;
+using System.IO;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    [Command(Name = "findpointersin", Help="Finds pointers to the GC heap within the given memory regions.")]
+    public class FindPointersInCommand : ExtensionCommandBase
+    {
+        private const string VtableConst = "vtable for ";
+
+        public IModuleService ModuleService { get; set; }
+        
+        public IMemoryRegionService MemoryRegionService { get; set; }
+        public ClrRuntime Runtime { get; set; }
+
+        [Option(Name = "--showAllObjects", Aliases = new string[] { "-a", "--all" }, Help = "Show all objects instead of only objects Pinned on the heap.")]
+        public bool ShowAllObjects { get; set; }
+
+        [Argument(Help = "The types of memory to search for pointers to the GC heap.")]
+        public string[] Regions { get; set; }
+
+        private int Width
+        {
+            get
+            {
+                int width = Console.WindowWidth;
+                if (width == 0)
+                    width = 120;
+                if (width > 256)
+                    width = 256;
+
+                return width;
+            }
+        }
+
+        public override void ExtensionInvoke()
+        {
+            if (Runtime is null)
```

Comment:
Since you are checking these services for null, the ServiceImport attribute needs Optional=true for each one them. You could also leave them no Optional and the command service will throw an error if it can't find the service. Then these checks wouldn't be needed.